roles/kubevirt_web_ui/tasks/start.yml

```diff
@@ -0,0 +1,46 @@
+---
+- name: Waiting for console rollout to complete
+  # `oc rollout status` will block until either the rollout succeeds or `spec.progressDeadlineSeconds` elapse.
+  # A zero return code indicates the rollout succeeded.
+  # https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#complete-deployment
+  command: >
+    {{ openshift_client_binary }} rollout status deployment/console --config={{ openshift.common.config_base }}/master/admin.kubeconfig -n kubevirt-web-ui
+  changed_when: false
+  # Ignore errors so we can log troubleshooting info on failures.
+  ignore_errors: yes
+  register: console_rollout_status
+
+# Log the result of `oc status`, `oc get pods`, `oc get events`, and `oc logs deployment/console` for troubleshooting failures.
+- when: console_rollout_status.rc != 0
```

Comment:
One concern I have with debugging inside the playbook is if a failure occurs in a debug task.  Even though you have ```ignore_errors: true```, a task can still log a failure, which can be misleading.

Instead of doing the debugging, you can print out a useful error message  then recommend  commands to debug.